### ISSUE TITLE
RequestablePaymentMethod event discrepancy at dropin creation between Firefox and Chrome

### ISSUE THREAD

**Issue by wagonli** [braintree/braintree-web-drop-in#511]
### General information

* SDK/Library version: 1.18.0
* Environment: Sandbox
* Browser and OS : Chome/Chromium on windows and fedora

### Issue description

When creating a dropin for a customer with saved payment method, we're seeing a difference between browsers :
- On Chrome a 'paymentMethodRequestable' event is raised when creating a drop-in with saved payment method (a default pre-existing payment method is selected)
- on Firefox, *no event* is raised when creating a drop-in with saved payment method (a default pre-existing payment method is selected)

**Comment by crookedneighbor** [comment]
Can you provide your client side code? I tried it in both Chrome and Firefox and both did _not_ fire the event when a saved payment method was pre-selected. I used the demo app (fill in the authorization query param with your client token): https://braintree.github.io/braintree-web-drop-in/?authorization=your-client-token-with-a-customer-id-here

Here's the code in the demo app for the listener: https://github.com/braintree/braintree-web-drop-in/blob/fffb91322adfe5604dfe7ceb05aac9912b9d8486/test/app/index.html#L231-L237

You can see that we specifically have to have logic for enabling the disabled submit button because it is not supposed to fire: https://github.com/braintree/braintree-web-drop-in/blob/fffb91322adfe5604dfe7ceb05aac9912b9d8486/test/app/index.html#L226-L229

**Comment by wagonli** [comment]
I cannot reproduce with the test app code.
I'll try to have a repro sample to give you.

Some difference that may have some impact that i'll check :
- `braintree-web-drop-in` dependency is loaded dynamically with webpack's `import('braintree-web-drop-in').then(/*create drop in*/)`
- I use angularjs to hide/show the drop in.

**Comment by wagonli** [comment]
No repro step here but the webpack dynamic import seems to imply that behavior on Chrome.
- Using a static import *fixes the issue* `import dropin from 'braintree-web-drop-in'` .
- Using dynamic 
``` 
var dropin = require('braintree-web-drop-in'); 
/*create drop in*/
```
*does not fix the issue*.
- Using AMD dynamic `require(['braintree-web-drop-in'], dropin => {/* create dropin */})` *does not fix the issue*

**Comment by crookedneighbor** [comment]
Do you have a simple Github Repo that reproduces the issue? Want to make sure I have the exact implementation that is causing the strange behavior.

**Comment by wagonli** [comment]
here you are : https://github.com/wagonli/braintree-issue-repro

**Comment by wagonli** [comment]
I see where the culprit might be, I trigger a `requestPaymentMethod` as soon as `requestablePaymentMethod` triggers. That may not be a good practice and with 3Dsecure activated it gives a pretty bad user interaction.

Nevertheless, there is still a discrepancy between firefox where I don't get any `requestablePaymentMethod` after dropin creation and Chrome where I get one.

**Comment by crookedneighbor** [comment]
Ahh, it has to do with the lazy importing that webpack can do. We'll just need a check to make sure that Drop-in is actually loaded before emitting events. Thanks for the report.

**Comment by crookedneighbor** [comment]
Fix is on master. Gonna re-open this until Drop-in is actually released with the fix.

**Comment by crookedneighbor** [comment]
This _should_ be fixed in v1.19.0, so I'm gonna close this. Report back here if it is still occurring.
